linode/manager
# Tech story [M3-6795]: Consolidate Link & ExternalLink

## Description 📝

### One component to rule them all! 💍 

This PR refactors the `Link` component so it can be used for all three use cases:
- a relative URL, which will render a `Link` component from React Router.
- an absolute, same domain/subdomain URL (ex: linode.com, akamai.com) that will render an `a` tag with `target="_blank"`.
- an absolute URL with the `external` prop, which will render an `a` tag with `target="_blank"` and an external link icon.


## Preview 📷
Visual changes should only pertain to the presence (or not) of some external links. Remember that links opening in a new tab but directing a user to an Akamai or Linode domain/subdomain should **not** feature an external link icon

## How to test 🧪
I can ask for reviewers to verify every link modified by this PR, however the areas to look for visual regressions or bad links should be:
1. Empty state landing pages
2. Docs Links on create pages (next to create buttons) (**note** - they are closer to the button now, which is intentional)
3. Help & Support area pages
4. Profile area
5. Storybook Link stories
